src/core/primitives.rs

```diff
@@ -41,6 +41,7 @@ fn compute_user_key_kmac(msk: &MasterSecretKey, usk: &UserSecretKey) -> Option<K
     user_key_bytes.extend_from_slice(&usk.a.to_bytes());
```

Comment:
I cannot comment on the exact line, but you are not allocating enough bytes to store Kyber keys if there are some. Plus, I would prefer all this to be inside a `map(|kmac_key| ...)` or a `if let Some(kmac_key) = &msk.kmac_key {` branche to avoid unneeded computation.

Reply:
to compute the right size to allocate we would need to iterate through the subkeys, is it worth it?
Maybe a "good enough" estimation for the size to allocate is the best solution here?

Reply:
solved by computing the KMAC directly instead of allocating a temporary vec